vm/default_runtime/src/lib.rs

```diff
@@ -546,3 +513,54 @@ fn transfer<ST: StateTree>(
 
     Ok(())
 }
+
+/// Returns public address of the specified actor address
+pub fn resolve_to_key_addr<'st, 'bs, ST, BS>(
+    st: &'st ST,
+    store: &'bs BS,
+    addr: &Address,
+) -> Result<Address, ActorError>
+where
+    ST: StateTree,
+    BS: BlockStore,
+{
+    if addr.protocol() == Protocol::BLS || addr.protocol() == Protocol::Secp256k1 {
+        return Ok(addr.clone());
+    }
+
+    let act = match st
+        .get_actor(&addr)
+        .map_err(|e| ActorError::new(ExitCode::SysErrInternal, e))?
+    {
+        Some(act) => act,
+        None => {
+            return Err(ActorError::new(
+                ExitCode::SysErrInternal,
+                format!("Failed to retrieve actor: {}", addr),
+            ))
+        }
+    };
```

Comment:
can you change this to be `.ok_or_else(|| ActorError::new( ...` instead of matching (as you have below), could make the message more descriptive that the actor was not found (optional though, this is fine)